store/src/androidTest/kotlin/com/dropbox/external/store5/ShareableLruCacheAndroidTests.kt

```diff
@@ -0,0 +1,230 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package com.dropbox.external.store5
+
+import com.dropbox.external.store5.fake.FakeNotes
+import com.dropbox.external.store5.fake.model.Note
+import com.dropbox.external.store5.impl.ShareableLruCache
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.last
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.junit.Before
+import org.junit.Test
+import kotlin.concurrent.thread
+import kotlin.test.assertEquals
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class ShareableLruCacheAndroidTests {
+    private val testScope = TestScope()
+    private lateinit var memoryLruCache: ShareableLruCache
+
+    @Before
+    fun before() {
+        memoryLruCache = ShareableLruCache(10)
+    }
+
+    private fun headPointer() = memoryLruCache.head
+    private fun tailPointer() = memoryLruCache.tail
+    private fun head() = memoryLruCache.head.next as ShareableLruCache.Node<Note>
```

Comment:
is this cast necessary? Wondering why it is not type safe?

Reply:
I can probably replace with `!!`

Reply:
Fixed!